### PR TITLE
CMR-7248 Not halting system when scheduler cannot start

### PR DESCRIPTION
**Pull request by jaybarra**
Adding better error handling to initializing a clean system bootstrapping by catching certain errors that occur when first running a system. This allows systems to come up when the database has not been initialized yet.

### CODE REVIEW

**Review by daniel-zamora — APPROVED** [review]

File: bootstrap-app/src/cmr/db.clj [hunk 1]
```diff
@@ -38,10 +38,10 @@
...
         (= "migrate" op)
         (drift/run
-         (conj
-          args
-          "-c" 
-          "config.bootstrap-migrate-config/app-migrate-config"))
+          (conj
+           (vec args)
+           "-c"
+           "config.bootstrap_migrate_config/app-migrate-config"))
```
Comment (mitchstartzel) [inline bootstrap-app/src/cmr/db.clj:44]: There are multiple changes just like this, but I don't see the matching namespaces changed to snake_case - why doesn't this break the function calls?

File: common-lib/src/cmr/common/jobs.clj [hunk 2]
```diff
@@ -179,24 +180,25 @@
...
-            ;; recreate the job at the same time. Sleeps between 0.5 seconds
-            ;; and 3 seconds.
-            (Thread/sleep (+ 500 (rand-int 2500)))
-            (recur (dec max-tries)))
-          (warn
-            (format
-              "All retries to schedule job [%s] failed." job-key)))))))
+    (async/go
+      (loop [max-tries 3]
+        (when-not (try-to-schedule-job scheduler job-key quartz-job trigger)
```
Comment (mitchstartzel) [inline common-lib/src/cmr/common/jobs.clj:190]: Can use `async/go-loop` here. Also, can we use a `<!` call in front of `try-to-schedule-job` for optimization?

File: dev-system/README.md [hunk 3]
```diff
@@ -79,8 +79,16 @@ $ docker build . -f Dockerfile.ecs
...
+```sh
+docker run \
+-p <SERVICE_PORT>:<SERVICE_PORT> \
+-e "CMR_URS_PASSWORD=<PASSWORD>" \
+-e "CMR_SYS_DBA_PASSWORD=<PASSWORD>" \
+-e "CMR_ORACLE_JAR_REPO=<URL_FOR_MAVEN_REPO>" \
+-e "CMR_INGEST_PASSWORD=<PASSWORD>" \
+-e "CMR_BOOTSTRAP_PASSWORD=<PASSWORD>" \
+-e "CMR_METADATA_DB_PASSWORD=<PASSWORD> \
+<IMAGE_ID> java -cp cmr-standalone.jar clojure.main -m cmr.<SERVICE_NAME>.runner
```
Comment (mitchstartzel) [inline dev-system/README.md:91]: :100:

**Review by mitchstartzel — APPROVED** [review]
Approved with a suggestion, and a question.

File: bootstrap-app/src/cmr/db.clj (hunk 1 above)
Comment (jaybarra) [reply bootstrap-app/src/cmr/db.clj:44]: These had never been changed and this is calling outside dependencies, which are named with underscores, but not translated into dashes. These classes never actually are called from within a running system

**Review by mschmele — APPROVED** [review]

**Pull request merged**